docs/markdown/Machine-files.md

```diff
@@ -5,24 +5,59 @@ documentation on the common values used by both, for the specific values of
 one or the other see the [cross compilation](Cross-compilation.md) and [native
 environments](Native-environments.md).
 
+## Data Types
+
+There are three basic data types in a machine file:
+- strings
+- arrays
+- booleans
+- integers
```

Comment:
There are three hard problems in programming:
- naming things
- off by one error